packages/trie/src/baseTrie.ts

```diff
@@ -65,13 +65,18 @@ export class Trie {
     this._deleteFromDB = deleteFromDB
 
     if (root) {
-      this.setRoot(root)
+      this.root = root
     }
   }
 
   /** Sets the current root of the `trie` */
   set root(value: Buffer) {
-    this.setRoot(value)
+    /* eslint-disable-next-line @typescript-eslint/no-unnecessary-condition */
```

Comment:
hm can you make `value` param optional with `value?: Buffer` so that it's not `no-unnecessary-condition`? (not sure if that's possible with the setter). another option would be to do `set root(value = this.empty_TRIE_ROOT)` and then no need for the `if (!value)` (I think I'd prefer this).

overall I like that you brought the `setRoot` logic here though, to dedupe the code then can we clear the code in `setRoot` and simply do `this.root = value`? I tend to like to use the new methods in the deprecated methods so they are easy to follow (and remove) in the future, and also if we update the set code we only have to do it in one place.

Reply:
> hm can you make `value` param optional with `value?: Buffer` so that it's not `no-unnecessary-condition`? (not sure if that's possible with the setter). another option would be to do `set root(value = this.empty_TRIE_ROOT)` and then no need for the `if (!value)` (I think I'd prefer this).
Typescript doesn't allow optional parameters in setters so I'm changing it to a union type with `undefined` to get around that and will update it my next commit.  Does that work?

Reply:
ah ok thanks, yeah let me add a follow up commit and I think it will be great. I will also add an entry to changelog.

Reply:
you know what, I changed my mind, I think your proposed solution was ultimately best, I'll change it back to ignoring the lint rule.